src/fides/api/ops/api/v1/endpoints/registration_endpoints.py

```diff
@@ -0,0 +1,81 @@
+import logging
+
+from fastapi import Depends
+from sqlalchemy.orm import Session
+from starlette.exceptions import HTTPException
+
+from fides.api.ops.api import deps
+from fides.api.ops.api.v1 import urn_registry as urls
+from fides.api.ops.models.registration import UserRegistration
+from fides.api.ops.schemas import registration as schemas
+from fides.api.ops.util.api_router import APIRouter
+
+router = APIRouter(
+    tags=["Registration"],
+    prefix=urls.V1_URL_PREFIX,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@router.get(
+    urls.REGISTRATION,
+    status_code=200,
+    response_model=schemas.GetRegistrationStatusResponse,
+)
+def get_registration_status(
+    *,
+    db: Session = Depends(deps.get_db),
+) -> schemas.GetRegistrationStatusResponse:
+    """
+    Return the registration status of this Fides deployment.
+    """
+    registrations = UserRegistration.all(db=db)
+    if registrations:
+        registration = registrations[0]
+        opt_in = registration.opt_in
+    else:
+        opt_in = False
```

Comment:
> This assumes there will only ever be a single registration. While that's intended, there's nothing forcing that to be true.

Correct, this is enforced by the API for now. I've explicitly added a test case for this reason. Granted, it's not ideal.

> It's inefficient to submit a query for all records on a table, regardless of the expected size of that table.

This is intentional so we can provide a meaningful error in the case where there is already a registration under a different `analytics_id`. Since there's only one record in the table, it's moot.

Reply:
> If this endpoint instead included an analytics_id in the URL path or request body, you could do:

`analytics_id` is in the data schema as a required field, so we have it accessible here without having to expose it via the URL. As above though, I've done it intentionally for error reporting as I think that's a more likely state to encounter.